### PR TITLE
Move tests from Mocha to AVA

### PR DESCRIPTION
**Pull request by schnittstabil**
Fixes: #10

### CODE REVIEW

File: test/async.js [hunk 1]
```diff
@@ -0,0 +1,220 @@
+import crypto from 'crypto';
+import fs from 'fs';
+import path from 'path';
+
```
Comment (kevva) [inline test/async.js]: Why are there empty lines between the imports?

File: test/async.js [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+const THREE_HUNDRED_KILO = (100 * 3 * 1024) + 1;
+
+test.before(() => {
+	process.chdir(path.dirname(__dirname));
+});
+
+test.beforeEach(t => {
+	const src = t.context.src = uuid.v4();
+	const dest = t.context.dest = uuid.v4();
+	t.context.creates = [src,	dest];
```
Comment (kevva) [inline test/async.js]: `t.context.creates = [src, dest];`, looks like you have a tab instead of a space there.

File: test/_assert.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
+'use strict';
+
```
Comment (kevva) [inline test/_assert.js]: Remove this line.

File: test/async.js [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+import pify from 'pify';
+import rewire from 'rewire';
+import rimraf from 'rimraf';
+import test from 'ava';
+import uuid from 'uuid';
+
+import cpFile from '../';
+import {assertDateEqual} from './_assert';
+
+const fsP = pify(fs, Promise);
```
Comment (kevva) [inline test/async.js]: `const fsP = pify(fs);`

File: test/async.js [hunk 5]
```diff
@@ -0,0 +1,220 @@
...
+import path from 'path';
+
+import bufferEquals from 'buffer-equals';
+import pify from 'pify';
+import rewire from 'rewire';
+import rimraf from 'rimraf';
+import test from 'ava';
+import uuid from 'uuid';
+
+import cpFile from '../';
```
Comment (kevva) [inline test/async.js]: `import m from '../';`

File: test/async.js [hunk 6]
```diff
@@ -0,0 +1,220 @@
...
+	const tmpStats = fs.lstatSync(t.context.dest);
+	assertDateEqual(licenseStats.atime, tmpStats.atime);
+	assertDateEqual(licenseStats.mtime, tmpStats.mtime);
+});
+
+test('throw an Error if `src` does not exists', async t => {
+	const err = await t.throws(cpFile('NO_ENTRY', t.context.dest));
+	t.is(err.name, 'CpFileError', 'wrong Error name: ' + err.stack);
+	t.is(err.code, 'ENOENT');
+	t.is(/`NO_ENTRY`/.test(err.message), true, 'Error message does not contain path: ' + err.message);
```
Comment (kevva) [inline test/async.js]: `t.true()`

File: test/async.js [hunk 7]
```diff
@@ -0,0 +1,220 @@
...
+	t.is(/`NO_ENTRY`/.test(err.stack), true, 'Error stack does not contain path: ' + err.stack);
```
Comment (kevva) [inline test/async.js]: Same as above.

File: test/async.js (hunk 1 above)
Comment (schnittstabil) [inline test/async.js]: Between system imports and package imports – should I drop them?

Comment (kevva) [inline test/async.js]: Yes

File: test/async.js (hunk 4 above)
Comment (schnittstabil) [inline test/async.js]: That is needed, as long as we support `v0.10`

File: test/async.js (hunk 5 above)
Comment (schnittstabil) [inline test/async.js]: Why `m`? @sindresorhus often uses `fn` afaik. – But it is not important to me…

File: test/async.js (hunk 6 above)
Comment (schnittstabil) [inline test/async.js]: Thank you – I think `t.regex` is even better...

File: test/_assert.js (hunk 3 above)
Comment (schnittstabil) [inline test/_assert.js]: This is on my list, but I want to do it after we drop `v0.10` support…

File: test/async.js (hunk 4 above)
Comment (kevva) [inline test/async.js]: You aren't using a `Promise` polyfill anywhere though. You're only passing the built in `Promise` (which is already used by `pify`).

File: test/async.js (hunk 5 above)
Comment (kevva) [inline test/async.js]: He used to, I think, but prefers `m` now.

File: test/async.js (hunk 6 above)
Comment (kevva) [inline test/async.js]: Yes :).

File: test/_assert.js (hunk 3 above)
Comment (kevva) [inline test/_assert.js]: I mean the empty line between `use strict` and the require.

File: test/async.js (hunk 4 above)
Comment (schnittstabil) [inline test/async.js]: It's not the build-in one, it's the Babel-polyfill `Promise`.
I've inserted it after `v0.10` tests fail…

**Comment by schnittstabil** [comment]
Ugh, took me some time to figure out, that with _reviews_ comments no longer become "outdated".
@kevva Only `const fsP = pify(fs, Promise);` is a bit arguable, the rest should be fixed.

File: test/_assert.js [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+ */
+exports.assertDateEqual = function (actual, expected, message) {
+	actual = new Date(actual);
+	expected = new Date(expected);
+
+	actual.setMilliseconds(0);
+	expected.setMilliseconds(0);
+
+	assert.equal(actual.getTime(), expected.getTime(), message);
+};
```
Comment (sindresorhus) [inline test/_assert.js]: This should no longer be needed now that we've dropped support for 0.10 and 0.12.

File: test/async.js [hunk 9]
```diff
@@ -0,0 +1,219 @@
...
+import pify from 'pify';
+import rewire from 'rewire';
+import rimraf from 'rimraf';
+import test from 'ava';
+import uuid from 'uuid';
+
+import m from '../';
+import {assertDateEqual} from './_assert';
+
+const fsP = pify(fs, Promise);
```
Comment (sindresorhus) [inline test/async.js]: No need to pass `Promise` anymore.

File: test/async.js [hunk 10]
```diff
@@ -0,0 +1,219 @@
+import crypto from 'crypto';
+import fs from 'fs';
+import path from 'path';
+import bufferEquals from 'buffer-equals';
+import pify from 'pify';
+import rewire from 'rewire';
+import rimraf from 'rimraf';
+import test from 'ava';
+import uuid from 'uuid';
+
```
Comment (sindresorhus) [inline test/async.js]: Remove this empty line

File: test/progress.js [hunk 11]
```diff
@@ -0,0 +1,69 @@
+import crypto from 'crypto';
+import fs from 'fs';
+import path from 'path';
+import rimraf from 'rimraf';
+import test from 'ava';
+import uuid from 'uuid';
+
```
Comment (sindresorhus) [inline test/progress.js]: Remove empty line

File: test/_assert.js (hunk 8 above)
Comment (schnittstabil) [inline test/_assert.js]: Unfortunately, `v4.6.0` and `v6.7.0` still need that :unamused:

**Comment by sindresorhus** [comment]
Looks great. Thank you @schnittstabil. 👌✨

**Pull request merged**